src/outputs/output_utils.cpp

```diff
@@ -316,11 +317,42 @@ std::size_t MPISum(std::size_t val) {
                 "size_t is unsigned and integral");
   static_assert(sizeof(std::size_t) == sizeof(unsigned long long int),
                 "MPI_UNSIGNED_LONG_LONG same as size_t");
+
+#endif
+}
+std::size_t MPISum(std::size_t val) {
+#ifdef MPI_PARALLEL
+  CheckMPISizeT();
   PARTHENON_MPI_CHECK(MPI_Allreduce(MPI_IN_PLACE, &val, 1, MPI_UNSIGNED_LONG_LONG,
                                     MPI_SUM, MPI_COMM_WORLD));
 #endif
   return val;
 }
 
+void CheckParameterInputConsistent(ParameterInput *pin) {
+#ifdef MPI_PARALLEL
+  CheckMPISizeT();
+
+  std::size_t pin_hash = std::hash<ParameterInput>()(*pin);
+  std::size_t pin_hash_root = pin_hash;
+  PARTHENON_MPI_CHECK(
+      MPI_Bcast(&pin_hash_root, 1, MPI_UNSIGNED_LONG_LONG, 0, MPI_COMM_WORLD));
+
+  int is_same_local = (pin_hash == pin_hash_root);
+  int pinput_same_accross_ranks;
+  PARTHENON_MPI_CHECK(MPI_Reduce(&is_same_local, &pinput_same_accross_ranks, 1, MPI_INT,
+                                 MPI_LAND, 0, MPI_COMM_WORLD));
+  if (Globals::my_rank == 0) {
+    PARTHENON_REQUIRE_THROWS(
+        pinput_same_accross_ranks,
+        "Parameter input object must be the same on every rank, otherwise I/O may "
+        "be\n\t\t"
+        "unable to write it safely. If you reached this error message, look to make "
+        "sure\n\t\t"
+        "that your calls to functions that look like pin->GetOrAdd are all called\n\t\t"
+        "exactly the same way on every MPI rank.");
+  }
```

Comment:
Don't think it matters, but isn't this reduction unnecessary since you could do the `PARTHENON_REQUIRE_THROWS` on each rank?

Reply:
yes that's true. Saving one reduce is probably worth it. I will change it.